core/sequencer/sequencer.go

```diff
@@ -239,12 +239,14 @@ func (s *Sequencer) applyMutations(mutations []*mutator.QueueMessage, leaves []*
 		}
 
 		// Serialize commitment.
-		extraData, err := proto.Marshal(m.ExtraData)
+		extraData, err := proto.Marshal(msg.ExtraData)
 		if err != nil {
 			glog.Warningf("Marshal(committed proto): %v", err)
 			continue
 		}
 
+		// Make sure that only ONE MapLeaf is output per index.
+		// Failure to do so will corrupt the map.
```

Comment:
"corrupt the map" is (I think) too strong -- maybe "... will result in a rejected request".

Reply:
Can you confirm that the mapserver will reject the request without processing it? 
Our previous theory was that duplicate leaves were creating errors deep in the map server.

Reply:
[It should do now](https://github.com/google/trillian/commit/be4e023c1707d2c6e836d97818d37501d910d5bb)

Reply:
Thanks